pkg/operatorsource/registryhelper.go

```diff
@@ -0,0 +1,37 @@
+package operatorsource
+
+import (
+	"context"
+
+	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/pkg/appregistry"
+
+	corev1 "k8s.io/api/core/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SetupRegistryOptions generates an Options object based on the OperatorSource spec. It passes along
+// the opsrc endpoint and, if defined, retrieves the authorization token from the specified Secret
+// object.
+func SetupRegistryOptions(client k8sclient.Client, spec *v1alpha1.OperatorSourceSpec) (appregistry.Options, error) {
+	options := appregistry.Options{
+		Source: spec.Endpoint,
+	}
+
+	auth := spec.AuthorizationToken
+	if auth.SecretName != "" && auth.SecretNamespace != "" {
+		secret := corev1.Secret{}
+		key := k8sclient.ObjectKey{
+			Name:      auth.SecretName,
+			Namespace: auth.SecretNamespace,
+		}
+		err := client.Get(context.TODO(), key, &secret)
+		if err != nil {
+			return options, err
+		}
+
+		options.AuthToken = string(secret.Data["token"])
```

Comment:
The auth token is already opaque, wondering whether we need to base64 encode it before we store it in the Secret object.

Reply:
@tkashem Is there a way to store it as a simple string without the encoding? I'm not too familiar with kube secrets, but my understanding is that the only real user data supported type is Opaque. Are you saying we should use the 'StringData' field to save the token in plain text? Won't that be harder to define later?